### PR TITLE
Filtered Itemlist + Map & Gridlist controls

### PR DESCRIPTION
**Pull request by PaulRitter**
items can now be hidden on the itemlist to avoid recreating the list on each filter input
makes the filter + list combo a standalone control things can inherit
creates map & gridlists to use in admin/mapping tooling
needs content pr (https://github.com/space-wizards/space-station-14/pull/8210) bc of the playerlist

### CODE REVIEW

File: Robust.Client/UserInterface/Controls/FilteredItemList.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿using Robust.Client.AutoGenerated;
+using Robust.Client.UserInterface.XAML;
+
+namespace Robust.Client.UserInterface.Controls;
+
+[GenerateTypedNameReferences, Virtual]
+public partial class FilteredItemList : BoxContainer
```
Comment (PJB3005) [inline Robust.Client/UserInterface/Controls/FilteredItemList.xaml.cs:7]: This control shouldn't publically inherit from `BoxContainer` directly. It should be a child control.

**Review by PJB3005 — COMMENTED** [review]
Is this intended for future mapping mode:tm: work? If so I'd prefer if these were internal or something.

We're keeping some facade of API stability now, leaving random classes with unclear use hanging around like this is sus.

**Comment by PaulRitter** [comment]
> Is this intended for future mapping mode™️ work? If so I'd prefer if these were internal or something.
> 
> We're keeping some facade of API stability now, leaving random classes with unclear use hanging around like this is sus.

I'd disagree on the unclear use. These can be used for mapping, but also for admin, hence they should be public imo.

**Comment by PJB3005** [comment]
There are concerns about how this would even work when PVS is enabled, since presumably we'd want to remove the hack of always synchronizing grids and maps eventually. How would this work then?

**Comment by PaulRitter** [comment]
> There are concerns about how this would even work when PVS is enabled, since presumably we'd want to remove the hack of always synchronizing grids and maps eventually. How would this work then?

we'll cross that bridge when we get there imo. there are multiple approaches to this, either networking grids seperately for these lists, or just sending all grids like we do now but to adminclients or smth

**Comment by moonheart08** [comment]
Sending all grids to admins would be bad for admin performance on servers that use very large numbers of grids simultaneously.

**Comment by PJB3005** [comment]
> either networking grids seperately for these lists

As soon as you go there these lists stop being any sort of "general purpose" solution and start being tied *heavily* to whatever UI/system (mapping mode etc) is using them. I do not think a blanket "list of all grids" control ever makes sense in that instance.

**Comment by metalgearsloth** [comment]
Paul's busy so.

**Pull request closed without merging**